packages/jest-mock-web-client/src/index.ts

```diff
@@ -0,0 +1,338 @@
+    },
+  };
+
+  /**
+   * groups method family
+   */
+  public readonly groups = {
+    archive: mockApi(),
+    create: mockApi(),
+    createChild: mockApi(),
+    history: mockApi(),
+    info: mockApi(),
+    invite: mockApi(),
+    kick: mockApi(),
+    leave: mockApi(),
+    list: mockApi(),
+    mark: mockApi(),
+    open: mockApi(),
+    rename: mockApi(),
+    replies: mockApi(),
+    setPurpose: mockApi(),
+    setTopic: mockApi(),
+    unarchive: mockApi(),
+  };
+
+  /**
+   * im method family
+   */
+  public readonly im = {
+    close: mockApi(),
+    history: mockApi(),
+    list: mockApi(),
+    mark: mockApi(),
+    open: mockApi(),
+    replies: mockApi(),
+  };
+
+  /**
+   * migration method family
+   */
+  public readonly migration = {
+    exchange: mockApi(),
+  };
+
+  /**
+   * mpim method family
+   */
+  public readonly mpim = {
+    close: mockApi(),
+    history: mockApi(),
+    list: mockApi(),
+    mark: mockApi(),
+    open: mockApi(),
+    replies: mockApi(),
+  };
+
+  /**
+   * oauth method family
+   */
+  public readonly oauth = {
+    access: mockApi(),
+    v2: {
+      access: mockApi(),
+    },
+  };
+
+  /**
+   * pins method family
+   */
+  public readonly pins = {
+    add: mockApi(),
+    list: mockApi(),
+    remove: mockApi(),
+  };
+
+  /**
+   * reactions method family
+   */
+  public readonly reactions = {
+    add: mockApi(),
+    get: mockApi(),
+    list: mockApi(),
+    remove: mockApi(),
+  };
+
+  /**
+   * reminders method family
+   */
+  public readonly reminders = {
+    add: mockApi(),
+    complete: mockApi(),
+    delete: mockApi(),
+    info: mockApi(),
+    list: mockApi(),
+  };
+
+  /**
+   * rtm method family
+   */
+  public readonly rtm = {
+    connect: mockApi(),
+    start: mockApi(),
+  };
+
+  /**
+   * search method family
+   */
+  public readonly search = {
+    all: mockApi(),
+    files: mockApi(),
+    messages: mockApi(),
+  };
+
+  /**
+   * stars method family
+   */
+  public readonly stars = {
+    add: mockApi(),
+    list: mockApi(),
+    remove: mockApi(),
+  };
+
+  /**
+   * team method family
+   */
+  public readonly team = {
+    accessLogs: mockApi(),
+    billableInfo: mockApi(),
+    info: mockApi(),
+    integrationLogs: mockApi(),
+    profile: {
+      get: mockApi(),
+    },
+  };
+
+  /**
+   * usergroups method family
+   */
+  public readonly usergroups = {
+    create: mockApi(),
+    disable: mockApi(),
+    enable: mockApi(),
+    list: mockApi(),
+    update: mockApi(),
+    users: {
+      list: mockApi(),
+      update: mockApi(),
+    },
+  };
+
+  /**
+   * users method family
+   */
+  public readonly users = {
+    conversations: mockApi(),
+    deletePhoto: mockApi(),
+    getPresence: mockApi(),
+    identity: mockApi(),
+    info: mockApi(),
+    list: mockApi(),
+    lookupByEmail: mockApi(),
+    setPhoto: mockApi(),
+    setPresence: mockApi(),
+    profile: {
+      get: mockApi(),
+      set: mockApi(),
+    },
+  };
```

Comment:
I know we talked about this a while back, but still might be worth at some point looking into mocking the APIs recursively. or maybe even generate parts of this file automatically, it's a lot to maintain

Reply:
I will give it another shot!

Reply:
Based on our conversation, my understanding is you gave this another shot and found the approach/existing code to be mocked wanting.